internal/pkg/service/stream/storage/level/local/encoding/encoder/csv/csv.go

```diff
@@ -11,13 +11,16 @@ import (
 	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/mapping/table"
 	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/mapping/table/column"
 	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/level/local/encoding/encoder/csv/fastcsv"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/level/local/encoding/encoder/result"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/level/local/encoding/writesync/notify"
 	"github.com/keboola/keboola-as-code/internal/pkg/utils/errors"
 )
 
 type Encoder struct {
 	columns     column.Columns
 	writersPool *fastcsv.WritersPool
 	valuesPool  *sync.Pool
+	notifier    func() *notify.Notifier
```

Comment:
Why is this (and many other parameters in the code) using type `func() *notify.Notifier` instead of just `*notify.Notifier`?

Reply:
If you pass the notifier as It was in my previous version on local, it would not be able to receive new notifier when `TriggerSync` was called. The function causes read lock to obtain notifier if it has changed based on previous `TriggerSync`.
I will use test sample `TestEncodingPipeline_Sync_Wait_ToDisk` 
```
write foo1
write foo1
wait for writting till it is completed (use notifier#1 from constructor)
triggersync (create new notifier#2)
```
but in next case
```
write foo2
wait for writing till it is completed (*use notifier#1 from constructor*)
trigger sync (create new notifier#3)
```
Therefore the code wouldn't work.
In new code we have this
```
write foo1
write foo1
wait for writting till it is completed (use notifier#1 from constructor)
triggersync (create new notifier#2)
```
next case
```
write foo2
wait for writing till it is completed (*use new notifier from triggersync*)
trigger sync (create new notifier#3)
```